### ISSUE TITLE
LibJS: Inline caching for property lookups

### ISSUE THREAD

**Issue by awesomekling** [SerenityOS/serenity#17984]
We should implement [inline caching](https://en.wikipedia.org/wiki/Inline_caching) for property lookups.

The basic idea is to use the shape system to cache the "location" of properties the first time we execute a non-computed `MemberExpression`, and then reuse that location on subsequent executions.

This isn't entirely trivial of course, since JavaScript is a dynamic language where you can replace somebody's prototype's prototype etc, so this does need some extra validation and deoptimization logic.

Basically:
- If a property turns out to be an own property, remember the current shape. If the shape hasn't changed next time we execute the same `MemberExpression`, we can find the property in the same location. See [this blog post](https://mathiasbynens.be/notes/shapes-ics) for inspiration.
- Caching prototype accesses is less straightforward. There are a number of approaches to this. See for example [this other blog post](https://mathiasbynens.be/notes/prototypes) that explains how V8 does this.
